pkgs/tools/misc/mkbootimage/default.nix

```
@@ -0,0 +1,40 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, libelf
+, pcre
+}:
+
+stdenv.mkDerivation rec {
+  pname = "mkbootimage";
+  version = "2.3dev";
+
+  src = fetchFromGitHub {
+    owner = "antmicro";
+    repo = "zynq-mkbootimage";
+    rev = "872363ce32c249f8278cf107bc6d3bdeb38d849f";
+    hash = "sha256-5FPyAhUWZDwHbqmp9J2ZXTmjaXPz+dzrJMolaNwADHs=";
+  };
+
+  propagatedBuildInputs = [ libelf pcre ];
+
+  patchPhase = ''
+    substituteInPlace Makefile --replace "git rev-parse --short HEAD" "echo nix"
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp mkbootimage $out/bin
```

nit:
```suggestion
    install -Dm755 mkbootimage -t $out/bin
```
You can also use the `install` command, although it may be a matter of preference.